src/Databases/DatabaseFilesystem.cpp

```diff
@@ -138,22 +138,28 @@ StoragePtr DatabaseFilesystem::getTableImpl(const String & name, ContextPtr cont
         return table;
 
     auto table_path = getTablePath(name);
-    if (!checkTableFilePath(table_path, context_, throw_on_error))
-        return {};
+    std::shared_ptr<ASTFunction> astFunctionPtr;
+    String format;
+    if (!containsGlobs(table_path))
+    {
+        if (!checkTableFilePath(table_path, context_, throw_on_error))
```

Comment:
We must check the path even if it contains globs. For example, we can allow globs only if the glob result is inside user_files directory, e.g. we need to check that glob symbols are only in allowed part of the path. Also queries like `SELECT * FROM '*.csv'` should only read from user_files directory as well. Let's make tests for these cases as well.

Reply:
> We must check the path even if it contains globs

Done!

> Also queries like SELECT * FROM '*.csv' should only read from user_files directory as well. Let's make tests for these cases as well.

`select * from '*csv'` works only in `clickhouse-local`, it allows to read from anywhere, not only from user_files, if I understand the code correctly.
in `clickhouse-client` It's allowed to use:
```sql
CREATE DATABASE test ENGINE = Filesystem('my_dir')
SELECT FROM test.<relative_path_to_test_folder> -- select * from test.`/../*some_file?{1..3}` will fail in clickhouse-client with PATH_ACCESS_DENIED
-- and
SELECT * FROM file(...)
```

I added more test cases for `clickhouse-client` and `clickhouse-local` `using ENGINE = Filesystem('my_dir').

Please tell me if there's anything else I need to do.